### PR TITLE
Consolidate into a central continuous integration workflow

### PR DESCRIPTION
**Pull request by peternied**
### Description
Previously there were different triggers for different workflows, moving all jobs together so they can share env variables for versions and whatnot.

🤞 That the code coverage numbers will sync up if there is a single workflow that pushes all the changes together consistantly.

### Testing
Validated the workflow running on my fork [1]

- [1] https://github.com/peternied/opensearch-migrations/actions/runs/9572832188/job/26393039644

### Check List
- [X] New functionality includes testing
  - [X] All tests pass, including unit test, integration test and doctest
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,4 +1,5 @@
 plugins {
+    id "com.dorongold.task-tree" version "2.1.1"
```
Comment (lewijacn) [inline build.gradle:2]: What does this do?

**Review by lewijacn — APPROVED** [review]
Glad to see these show as separate jobs on the PR page and feels a bit cleaner 👍 . I am a little concerned that having each job have to upload its coverage report might retain the codecoverage discrepancies we already have (no proof of this), but am hoping that the actual issue is that codecoverage needs us to upload every coverage report for all our jobs on each PR, and this PR should enable that

**Review by AndreKurait — APPROVED** [review]

File: build.gradle (hunk 1 above)
Comment (peternied) [reply build.gradle:2]: Oops, didn't mean to include that but actually - its pretty nice for troubleshooting gradle dependencies - specifically it shows what dependencies TASKS have, e.g.:

```
ubuntu@ip-172-31-18-157:~/git/opensearch-migrations$ ./gradlew -p RFS assemble taskTree

> Task :RFS:taskTree

------------------------------------------------------------
Project ':RFS'
------------------------------------------------------------

:RFS:assemble
+--- :RFS:distTar
|    +--- :commonDependencyVersionConstraints:jar
|    |    +--- :commonDependencyVersionConstraints:classes
|    |    |    +--- :commonDependencyVersionConstraints:compileJava
|    |    |    \--- :commonDependencyVersionConstraints:processResources
|    |    \--- :commonDependencyVersionConstraints:compileJava *
|    +--- :coreUtilities:jar
|    |    +--- :coreUtilities:classes
|    |    |    +--- :coreUtilities:compileJava
|    |    |    |    +--- :commonDependencyVersionConstraints:compileJava *
|    |    |    |    \--- :coreUtilities:generateEffectiveLombokConfig
|    |    |    \--- :coreUtilities:processResources
|    |    \--- :coreUtilities:compileJava *
|    +--- :RFS:jar
|    |    +--- :RFS:classes
|    |    |    +--- :RFS:compileJava
|    |    |    |    +--- :commonDependencyVersionConstraints:compileJava *
|    |    |    |    +--- :coreUtilities:compileJava *
|    |    |    |    \--- :RFS:generateEffectiveLombokConfig
|    |    |    \--- :RFS:processResources
|    |    \--- :RFS:compileJava *
|    \--- :RFS:startScripts
|         +--- :commonDependencyVersionConstraints:jar *
|         +--- :coreUtilities:jar *
|         \--- :RFS:jar *
+--- :RFS:distZip
|    +--- :commonDependencyVersionConstraints:jar *
|    +--- :coreUtilities:jar *
|    +--- :RFS:jar *
|    \--- :RFS:startScripts *
\--- :RFS:jar *


(*) - subtree omitted (printed previously)
Add --repeat to allow printing a subtree of the same task more than once.

To see task dependency tree for a specific task, run gradlew <project-path>:<task> <project-path>:taskTree [--depth <depth>] [--with-inputs] [--with-outputs] [--repeat]
Executions of all tasks except for taskTree are skipped. They are used for building the task graph only.
For example, try running gradlew :RFS:build :RFS:taskTree

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/8.0.2/userguide/command_line_interface.html#sec:command_line_warnings
```

Comment (peternied) [reply build.gradle:2]: I'm going to keep it in - if anyone would prefer we remove it happy to do so in a follow up PR

**Pull request merged**